File: messages/recover_lt.js
const fs = require('fs');
const path = '/Users/umurey/.gemini/antigravity/brain/77e8e05b-cc51-4c15-aea9-7c53a358f3dd/.system_generated/logs/transcript.jsonl';
let lt_text = fs.readFileSync('lt.json', 'utf-8');

const lines = fs.readFileSync(path, 'utf-8').split('\n');
for (const line of lines) {
    if (!line) continue;
    try {
        const entry = JSON.parse(line);
        if (entry.type === 'PLANNER_RESPONSE' && entry.tool_calls) {
            for (const call of entry.tool_calls) {
                if (call.name === 'multi_replace_file_content' || call.name === 'replace_file_content') {
                    const args = call.args;
                    let targetFile = args.TargetFile || args.targetFile || "";
                    if (targetFile.includes('lt.json')) {
                        let chunks = args.ReplacementChunks || args.replacementChunks || [];
                        if (typeof chunks === 'string') {
                            try {
                                // Fix unescaped newlines inside the JSON string
                                chunks = chunks.replace(/\n/g, '\\n').replace(/\r/g, '\\r').replace(/\t/g, '\\t');
                                chunks = JSON.parse(chunks);
                            } catch(e) {
                                console.log("Error parsing chunk string", e.message);
                                continue;
                            }
                        }
                        
                        if (call.name === 'replace_file_content') {
                            chunks = [args];
                        }

                        for (const chunk of chunks) {
                            const target = chunk.TargetContent || chunk.targetContent || "";
                            const repl = chunk.ReplacementContent || chunk.replacementContent || "";
                            if (target && lt_text.includes(target)) {
                                lt_text = lt_text.replace(target, repl);
                            }
                        }
                    }
                }
            }
        }
    } catch(e) {
        console.log("Error line", e.message);
    }
}
fs.writeFileSync('lt_recovered.json', lt_text);
console.log("Recovery done");

File: messages/recover_lt.py
import json

def apply_recovery():
    transcript_path = "/Users/umurey/.gemini/antigravity/brain/77e8e05b-cc51-4c15-aea9-7c53a358f3dd/.system_generated/logs/transcript.jsonl"
    with open("lt.json", "r") as f:
        lt_text = f.read()

    lines = open(transcript_path).readlines()
    for line in lines:
        try:
            entry = json.loads(line)
        except: continue
        
        if entry.get("type") == "PLANNER_RESPONSE":
            calls = entry.get("tool_calls", [])
            for call in calls:
                if call.get("name") == "multi_replace_file_content":
                    args = call.get("args", {})
                    target_file = args.get("TargetFile", "")
                    if "lt.json" in target_file:
                        chunks = args.get("ReplacementChunks", [])
                        if isinstance(chunks, str):
                            try:
                                chunks = json.loads(chunks, strict=False)
                            except Exception as e:
                                print(f"Error parsing chunks: {e}")
                                continue
                        
                        for chunk in chunks:
                            target = chunk.get("TargetContent", "")
                            repl = chunk.get("ReplacementContent", "")
                            if target and target in lt_text:
                                lt_text = lt_text.replace(target, repl)
                                
    with open("lt_recovered.json", "w") as f:
        f.write(lt_text)

apply_recovery()
print("Recovery completed.")
